### PR TITLE
don't require go.sum in go-check workflow

### LINKED ISSUE #73: go-check workflow fails if the repo doesn't have a go.sum
... and apparently it only needs one if there are entries in `go.sum`.

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #73.

If `go.mod` doesn't contain any dependencies, it's not necessary to have a `go.sum` file.

Unfortunately, this change will trigger a PR in all repos that we've deployed to so far.

### CODE REVIEW

**Review by Stebalien — APPROVED** [review]

**Comment by marten-seemann** [comment]
Waiting for @mvdan's approval before merging. He's had valuable input for bash and git in the past.

**Review by mvdan — APPROVED** [review]
I'm pretty sure `tidy` always adds a go.sum these days, even if it's empty. I assume this is about not failing in `cp go.sum go.sum.orig` if the file doesn't exist in the main branch yet.

**Comment by marten-seemann** [comment]
>  I assume this is about not failing in cp go.sum go.sum.orig if the file doesn't exist in the main branch yet.

That's correct.

> I'm pretty sure tidy always adds a go.sum these days, even if it's empty.

It doesn't.

**Comment by mvdan** [comment]
> It doesn't.

Huh, you're right, I'm not sure where I got that idea from.

**Pull request merged**
